tinker_cookbook/renderers.py

```diff
@@ -1383,25 +1574,34 @@ def render_message(self, message: Message, ctx: RenderContext) -> RenderedMessag
                 - is_last: Whether this is the last message (affects thinking stripping)
                 - prev_message: The previous message, used to detect post-tool formatting
         """
-        assert message.get("thinking") is None, "TODO: support thinking attribute in DSV3 renderer"
-        assert isinstance(message["content"], str), (
-            "DeepSeekV3ThinkingRenderer only supports message with string content"
-        )
-
         # Check if this assistant message follows a tool response
         follows_tool = ctx.prev_message is not None and ctx.prev_message["role"] == "tool"
 
+        # Check if content has ThinkingPart in list form for cleaner stripping
+        content = message["content"]
+        has_thinking_part = (
+            isinstance(content, list)
+            and any(p["type"] == "thinking" for p in content)
+        )
+
+        # Only use ensure_text for non-ThinkingPart content
+        # (ThinkingPart content is handled separately in the assistant branch)
+        if not has_thinking_part:
+            content_str = ensure_text(content)
+        else:
+            content_str = ""  # Will be set in assistant branch
```

Comment:
this seems fragile. can you refactor, or set this to some value that'll fail more loudly, if code gets changed later and we forget to set it?